xgi/core/hypergraph.py

```diff
@@ -958,6 +959,86 @@ def double_edge_swap(self, n_id1, n_id2, e_id1, e_id2):
         self._edge[e_id1] = temp_members1
         self._edge[e_id2] = temp_members2
 
+    def random_edge_shuffle(self, e_id1=None, e_id2=None):
+        """Randomly redistributes nodes between two hyperedges.
+
+        The process is as follows:
+
+        1. randomly select two hyperedges
+        2. place all their nodes into a single bucket
+        3. randomly redistribute the nodes between those two hyperedges
+
+        Parameters
+        ----------
+        e_id1 : node ID, optional
+            ID of first edge to shuffle.
+        e_id2 : node ID, optional
+            ID of second edge to shuffle.
+
+        Note
+        ----
+        After shuffling, the sizes of the two hyperedges are unchanged.
+        Edge IDs and attributes are also unchanged.
+        If the same node appears in both hyperedges, then this is still true after reshuffling.
+        If either `e_id1` or `e_id2` is not provided, then two random edges are selected.
+
+        Reference
+        ---------
+        Philip S C., 2020.
+        "Configuration models of random hypergraphs."
+        Journal of Complex Networks, 8(3).
+        https://doi.org/10.1093/comnet/cnaa018
+
+        Example
+        -------
+        >>> import xgi
+        >>> random.seed(42)
+        >>> H = xgi.Hypergraph([[1, 2, 3], [3, 4], [4, 5]])
+        >>> H.random_edge_shuffle()
+        >>> H.edges.members()
+        [{2, 4, 5}, {3, 4}, {1, 3}]
+
+        """
+        if len(self._edge) < 2:
+            raise ValueError("Hypergraph must have at least two edges.")
+
+        # select two random edges
+        if e_id1 is None or e_id2 is None:
+            e_id1, e_id2 = random.sample(list(self._edge), 2)
+
+        # get nodes in those edges
+        nodes_e1 = self._edge[e_id1]
```

Comment:
I don't care either way on this, but perhaps you can eliminate the `node_` prefix for some of these variable names for brevity?

Reply:
@nwlandry OK, I guess "edges are node lists" is self-explanatory. Fixed in a286f66.